### PR TITLE
dev/core#2556 Rename extension org.civicrm.search -> org.civicrm.search_kit

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Rename extension to avoid a namespace conflict in Drupal 7 (and it's a more descriptive name anyway).
See https://lab.civicrm.org/dev/core/-/issues/2556

### CODE REVIEW

File: ext/search_kit/sql/auto_install.sql [hunk 1]
```diff
@@ -64,6 +63,6 @@ CREATE TABLE `civicrm_search_display` (
   )
   
 ,          CONSTRAINT FK_civicrm_search_display_saved_search_id FOREIGN KEY (`saved_search_id`) REFERENCES `civicrm_saved_search`(`id`) ON DELETE CASCADE  
-) ROW_FORMAT=DYNAMIC    ;
```
Comment (colemanw) [inline ext/search_kit/sql/auto_install.sql:67]: @seamuslee001 this is what happens when I rerun civix :(

Comment (colemanw) [reply ext/search_kit/sql/auto_install.sql:67]: I've manually re-added the `ROW_FORMAT=DYNAMIC` bit.

**Comment by demeritcowboy** [comment]
I tested an upgrade and it worked but I got the error popup you get when you've deleted an installed extension's folder and I had to clear cache manually after the upgrade for it to work again. Maybe something about the order of when menu paths get rebuilt.

**Comment by colemanw** [comment]
Thanks for testing @demeritcowboy. IMO it would certainly be _nice_ to fix that issue, but not necessarily a blocker to merging this, as the extra cache flush isn't the worst thing in the world for the handful of sites using the extension (it's officially still beta).

**Comment by colemanw** [comment]
@demeritcowboy I think I found a fix - pushed a second commit c486a70 which results in a better experience. In my testing, the error about a mising extension still pops up once after the upgrade, but then goes away by itself without a manual cache clear, and the search kit page works right away (previously it couldn't be found prior to flushing caches several times).

**Comment by demeritcowboy** [comment]
Cool. Will take a look.

**Comment by colemanw** [comment]
@totten I'm a little surprised the upgrader doesn't flush the extension cache per c486a70 as a standard part of the upgrade.

**Comment by demeritcowboy** [comment]
Ok thanks retested and yes it's better. Didn't even get any popups.
I didn't test whether search kit still fully works, just a quickie contact test.

**Comment by seamuslee001** [comment]
I'm going to merge based on Dave D's testing

**Comment by colemanw** [comment]
Thanks @demeritcowboy !

**Pull request merged**